### PR TITLE
feat(react-grid): add components for the VirtualTable plugin

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

File: packages/dx-react-grid/docs/reference/virtual-table.md [hunk 1]
```diff
@@ -17,6 +17,19 @@ estimatedRowHeight | number | `37` for [Bootstrap3](https://www.npmjs.com/packag
 
 This plugin also supports the [Table](table.md#properties) plugin's properties.
 
+## Plugin Components
```
Comment (kvet) [inline packages/dx-react-grid/docs/reference/virtual-table.md]: I don't like an idea to duplicate docs for VirtualTable. Can we avoid it?

Comment (MaximKudriavtsev) [reply packages/dx-react-grid/docs/reference/virtual-table.md]: @kvet  Maybe we must write the next sentence?
"You can use all Table's components from VirtualTable. For example VirtualTable.Cell"

Comment (kvet) [reply packages/dx-react-grid/docs/reference/virtual-table.md]: Yes, it's way better. @viterobk what do you think?

Comment (viterobk) [reply packages/dx-react-grid/docs/reference/virtual-table.md]: I'd better duplicate all the Table plugin's descriptions in Virtual Table topic. "Same as Table's" could be appropriate if Virtual Table actually extended Table. However, it simply has similar properties, dependencies, imports an exports.

File: packages/dx-react-grid/docs/reference/virtual-table.md [hunk 2]
```diff
@@ -1,28 +1,68 @@
...
+- [DataTypeProvider](data-type-provider.md) [Optional]
 
 ### Properties
 
 Name | Type | Default | Description
 -----|------|---------|------------
 height | number | 530 | The virtual table's height.
 estimatedRowHeight | number | `37` for [Bootstrap3](https://www.npmjs.com/package/@devexpress/dx-react-grid-bootstrap3); `48` for [Material UI](https://www.npmjs.com/package/@devexpress/dx-react-grid-material-ui) | An estimated value of the row height. For a table whose rows have a variable height, specify an average value. The more accurately you estimate the row height, the better the virtual table performs.
+layoutComponent | ElementType&lt;[TableLayoutProps](table.md#tablelayoutprops)&gt; | | A component that renders a table layout.
+getCellComponent | (columnName: string) => ElementType&lt;[TableDataCellProps](#tabledatacellprops)&gt; | | A function returning a component that renders a table cell for a specific column.
```
Comment (kvet) [inline packages/dx-react-grid/docs/reference/virtual-table.md]: Where is the table.md reference?

File: packages/dx-react-grid/docs/reference/virtual-table.md [hunk 3]
```diff
@@ -1,28 +1,68 @@
...
+columns | Getter | Array&lt;[Column](#column-extension)&gt; | Columns to be rendered by the virtual table view.
+getRowId | Getter | (row: any) => number &#124; string | A function used to get a unique row identifier.
+getCellValue | Getter | (row: any, columnName: string) => any | A function used to get the column value for a given row.
 
 ### Exports
 
-Same as the [Table](table.md#exports)'s.
+Name | Plugin | Type | Description
+-----|--------|------|------------
+tableHeaderRows | Getter | Array&lt;[TableRow](#tablerow)&gt; | Header rows to be rendered.
```
Comment (kvet) [inline packages/dx-react-grid/docs/reference/virtual-table.md]: The same here

File: packages/dx-react-grid/docs/reference/virtual-table.md [hunk 4]
```diff
@@ -1,28 +1,68 @@
...
 
 ### Imports
 
-Same as the [Table](table.md#imports)'s.
+Name | Plugin | Type | Description
+-----|--------|------|------------
+rows | Getter | Array&lt;any&gt; | Rows to be rendered by the virtual table view.
+columns | Getter | Array&lt;[Column](table.md#column-extension)&gt; | Columns the virtual table view should render.
+getRowId | Getter | (row: any) => number &#124; string | A function used to get a unique row identifier.
+getCellValue | Getter | (row: any, columnName: string) => any | A function used to get the column value for a given row.
```
Comment (dirk-pieterse) [inline packages/dx-react-grid/docs/reference/virtual-table.md]: getCellValue -> gets the column value for a row? The name and description don't to match.

**Pull request merged**